src/modal/modal.scss

```diff
@@ -74,3 +82,34 @@
     transition-delay: 0s;
   }
 }
+
+@mixin iui-modal-full-page {
+  --iui-full-page-modal-transform-x: 100%;
+  --iui-full-page-modal-transform-y: 0%;
+
+  visibility: visible;
+  opacity: 1;
+  transform: translate(var(--iui-full-page-modal-transform-x), var(--iui-full-page-modal-transform-y));
```

Comment:
Can we try using `scale(0)` to `scale(1)` instead of animating in from the side?  When it animates in from the right it makes me think we are making progress (moving forward in a step like a wizard) which may not always be the case.  If you use scale you'll need to transition the transform, opacity, and visibility.  I think using scale will also allow you to remove lines 90-91.

Reply:
Agree. Also maybe it's good idea to use dark background overlay (same as in not full page modals) to show users what is happening. Currently it's hard to see modal border when it's appearing from the side.

Reply:
I would have thought it should animate from the bottom. That's what I'm used to seeing everywhere.

But also this animation will not work in React. Currently it relies on the dialog being present (and hidden) in the DOM, whereas in React, there isn't a distinction between "hidden" or "visible": it's either present in the DOM (when open). or it isn't (when closed).

Reply:
I added separate mixins for slide animations vs scale animations. Slide animations now default from the bottom. I removed
`visibility: visible;
opacity: 1;`
from the slide class and am now dealing with it in the transition.

Reply:
I tried to add the dark background to the full page modal during the transitions, but it's more complicated than I anticipated and need to think about it more. The dark background is currently being covered by the full page modal and moving with it (slide or scale), so I'm not sure how to have that part remain constant without adding another HTML element.

Reply:
I added the dark background. There are new videos in the PR description.